### PR TITLE
feat: AuthZ Hub-KMS basic integration

### PR DESCRIPTION
**Pull request by rolsonquadras**
- Create user secret
- Split the user secret
- Integrate AuthZ key server to create the keystore

part of #485 

Signed-off-by: Rolson Quadras <rolson.quadras@securekey.com>

### CODE REVIEW

File: pkg/restapi/oidc/operations.go [hunk 1]
```diff
@@ -386,3 +411,70 @@ func (o *Operation) userLogoutHandler(w http.ResponseWriter, r *http.Request) {
...
+	}
+
+	secrets, err := o.secretSplitter.Split(b, 2, 2)
+	if err != nil {
+		return fmt.Errorf("split user secret key : %w", err)
+	}
+
+	// TODO https://github.com/trustbloc/edge-agent/issues/488 send half secret key to hub-auth and remove logger
+	logger.Errorf(string(secrets[0]))
+	logger.Errorf(string(secrets[0]))
```
Comment (llorllale) [inline pkg/restapi/oidc/operations.go:430]: nit: logging the same share twice

Comment (llorllale) [reply pkg/restapi/oidc/operations.go:430]: also, log level

File: pkg/restapi/oidc/operations.go [hunk 2]
```diff
@@ -386,3 +411,70 @@ func (o *Operation) userLogoutHandler(w http.ResponseWriter, r *http.Request) {
...
+
+	reqBytes, err := json.Marshal(createKeystoreReq{
+		Controller: sub,
+	})
+	if err != nil {
+		return fmt.Errorf("marshal create keystore req : %w", err)
+	}
+
+	req, err := http.NewRequestWithContext(context.TODO(),
```
Comment (llorllale) [inline pkg/restapi/oidc/operations.go:439]: access_token added in followup?

Comment (rolsonquadras) [reply pkg/restapi/oidc/operations.go:439]: Yes, updated the TODO [comment](https://github.com/trustbloc/edge-agent/pull/490/files#diff-96466bce0ffd4cb8bd674beee84455ded8463ad6bdf53e49635ef6c333eddc66R445).

File: pkg/restapi/oidc/operations.go (hunk 1 above)
Comment (rolsonquadras) [reply pkg/restapi/oidc/operations.go]: updated

**Review by sudeshrshetty — APPROVED** [review]

**Review by llorllale — APPROVED** [review]

**Pull request merged**